compiler/rustc_hir_analysis/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -435,6 +435,22 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         &self,
         num_params_to_take: usize,
     ) -> String {
+        let is_in_a_method_call = self
+            .tcx
+            .hir()
+            .parent_iter(self.path_segment.hir_id)
+            .skip(1)
+            .find_map(|(_, node)| match node {
+                hir::Node::Expr(expr) => Some(expr),
+                _ => None,
+            })
+            .is_some_and(|expr| {
+                matches!(
+                    expr.kind,
+                    hir::ExprKind::MethodCall(hir::PathSegment { args: Some(_), .. }, ..)
+                )
+            });
```

Comment:
Isn't this going to be true when we're inside a nested item? You should probably limit this search somehow so it doesn't trigger for like

```
fn hello() {
  x.foo({
    fn bar() {
        let x: Vec;
    }
  });

Reply:
`find_map` picks only the first expression, so it stops at the block around `fn`. I don't think it's possible to have an item directly as an argument.

I've considered checking whether `MethodCall`'s `args` contain `self.path_segment`, but that requires digging a few layers deep, and the code for it is pretty involved.

BTW, `let x: Vec<T>` is not a great suggestion, and `let x: Vec<_>` would have been better. I'm wondering whether it should always use `_` on non-items inside function bodies, and limit suggestions of parameter names to items where `_` is not allowed.